src/codegen/sys/lib_.rs

```diff
@@ -232,10 +233,23 @@ fn prepare_enum_member_name(name: &str) -> String {
     }
 }
 
-fn generate_classes_structs(w: &mut Write, classes: &[&Class]) -> Result<()> {
+fn generate_classes_structs(w: &mut Write, env: &Env, classes: &[&Class]) -> Result<()> {
     try!(writeln!(w, ""));
     for klass in classes {
-        try!(writeln!(w, "#[repr(C)]\npub struct {}(c_void);", klass.c_type));
+        let (lines, commented) = generate_fields(env, &klass.name, &klass.fields);
+        
+        let comment = if commented { "//" } else { "" };
+        if lines.is_empty() {
+            try!(writeln!(w, "{}#[repr(C)]\n{0}pub struct {}(c_void);\n", comment, klass.c_type));
+        }
+        else {
+            try!(writeln!(w, "{}#[repr(C)]\n{0}pub struct {} {{", comment, klass.c_type));
+
+            for line in lines {
+                try!(writeln!(w, "{}{}", comment, line));
```

Comment:
Also, the original code wasn't written by me (I think @EPashkin wrote it), I simply modified it to work with classes as well as records, as I didn't see any reason it shouldn't.